test_infra/src/lib.rs

```diff
@@ -3,21 +3,65 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#[macro_use]
+extern crate lazy_static;
+
 use ssh2::Session;
+use std::env;
+use std::ffi::OsStr;
 use std::fs;
 use std::io;
 use std::io::{Read, Write};
 use std::net::TcpListener;
 use std::net::TcpStream;
 use std::os::unix::io::AsRawFd;
-use std::path::Path;
-use std::process::{ExitStatus, Output};
+use std::path::{Path, PathBuf};
+use std::process::{Child, Command, ExitStatus, Output, Stdio};
 use std::str::FromStr;
+use std::sync::Mutex;
 use std::thread;
 use vmm_sys_util::tempdir::TempDir;
 
 pub const DEFAULT_TCP_LISTENER_MESSAGE: &str = "booted";
 
+pub const DIRECT_KERNEL_BOOT_CMDLINE: &str =
+    "root=/dev/vda1 console=hvc0 rw systemd.journald.forward_to_console=1";
+
+pub const PIPE_SIZE: i32 = 32 << 20;
+
+#[cfg(target_arch = "x86_64")]
+pub mod x86_64 {
+    pub const BIONIC_IMAGE_NAME: &str = "bionic-server-cloudimg-amd64.raw";
+    pub const FOCAL_IMAGE_NAME: &str = "focal-server-cloudimg-amd64-custom-20210609-0.raw";
+    pub const FOCAL_SGX_IMAGE_NAME: &str = "focal-server-cloudimg-amd64-sgx.raw";
+    pub const HIRSUTE_NVIDIA_IMAGE_NAME: &str = "hirsute-server-cloudimg-amd64-nvidia.raw";
+    pub const FOCAL_IMAGE_NAME_QCOW2: &str = "focal-server-cloudimg-amd64-custom-20210609-0.qcow2";
+    pub const FOCAL_IMAGE_NAME_VHD: &str = "focal-server-cloudimg-amd64-custom-20210609-0.vhd";
+    pub const FOCAL_IMAGE_NAME_VHDX: &str = "focal-server-cloudimg-amd64-custom-20210609-0.vhdx";
+    pub const WINDOWS_IMAGE_NAME: &str = "windows-server-2019.raw";
+    pub const OVMF_NAME: &str = "OVMF.fd";
+    pub const GREP_SERIAL_IRQ_CMD: &str = "grep -c 'IO-APIC.*ttyS0' /proc/interrupts || true";
+}
+
+#[cfg(target_arch = "x86_64")]
+use x86_64::*;
+
+#[cfg(target_arch = "aarch64")]
+pub mod aarch64 {
+    pub const BIONIC_IMAGE_NAME: &str = "bionic-server-cloudimg-arm64.raw";
+    pub const FOCAL_IMAGE_NAME: &str = "focal-server-cloudimg-arm64-custom-20210929-0.raw";
+    pub const FOCAL_IMAGE_UPDATE_KERNEL_NAME: &str =
+        "focal-server-cloudimg-arm64-custom-20210929-0-update-kernel.raw";
+    pub const FOCAL_IMAGE_NAME_QCOW2: &str = "focal-server-cloudimg-arm64-custom-20210929-0.qcow2";
+    pub const FOCAL_IMAGE_NAME_VHD: &str = "focal-server-cloudimg-arm64-custom-20210929-0.vhd";
+    pub const FOCAL_IMAGE_NAME_VHDX: &str = "focal-server-cloudimg-arm64-custom-20210929-0.vhdx";
+    pub const GREP_SERIAL_IRQ_CMD: &str = "grep -c 'GICv3.*uart-pl011' /proc/interrupts || true";
+    pub const GREP_PMU_IRQ_CMD: &str = "grep -c 'GICv3.*arm-pmu' /proc/interrupts || true";
+}
```

Comment:
Do you need to move these in the first place?

Reply:
It was to make sure that every (the first) commit can be compiled correctly. If that's not something we want to ensure, I can merge the two "moves". Please let me know your thoughts.

Reply:
We want that, I just thought you didn't need to move these const.

Reply:
Got it. Just realize that I can try to do the refactoring on the `fw_path()` first, and that should allow us to avoid the extra moves. :)